src/types.rs

```diff
@@ -86,8 +86,8 @@ macro_rules! new_type {
             $($item)*
 
             #[doc = $new_doc]
-            pub const fn new(s: $type) -> Self {
-                $name(s)
+            pub fn new(s: impl Into<$type>) -> Self {
```

Comment:
I'm realizing that having `const` here probably didn't add much value since `String`s can't be const initialized 👍